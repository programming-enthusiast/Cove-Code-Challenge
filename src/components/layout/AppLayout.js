import './style.css'
import React from 'react'
import Sidebar from './sidebar/Sidebar'

export default function AppLayout({ children }) {
  return (
    <div className="app-layout__container">
      <Sidebar/>
      <div className="app-layout__page-container">
        {children}
      </div>
    </div>
  )
}
