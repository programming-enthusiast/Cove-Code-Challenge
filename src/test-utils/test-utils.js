import React from "react";
import { createMemoryHistory } from "history";
import { Router, Route } from "react-router-dom";
import renderer from "react-test-renderer"

export function renderWithProviders(
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {},
  apolloMocks
) {
  console.log("I am route inside renderWithProviders Wrapper:", route);
  return {
    ...renderer.create(
      <Router history={history}>
        <Route path={route}>{ui}</Route>
      </Router>
    ),
    history
  };
}
