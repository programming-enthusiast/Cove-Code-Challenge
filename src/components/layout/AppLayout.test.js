import React from "react"
import { Route } from "react-router-dom";
import AppLayout from "./AppLayout"
import { renderWithProviders } from "../../test-utils/test-utils";


describe("AppLayout Component", () => {
  let renderedComponent

  beforeAll(() => {
    renderedComponent = renderWithProviders(
      <Route path="/">
        <AppLayout/>
      </Route>,
      {
        route: "/"
      })
  })

  it("should render correctly", () => {
    expect(renderedComponent.toJSON()).toMatchSnapshot();
  })
})
