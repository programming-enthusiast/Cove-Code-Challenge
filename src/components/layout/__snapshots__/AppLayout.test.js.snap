// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout Component should render correctly 1`] = `
<div
  className="app-layout__container"
>
  <div
    className="sidebar__container"
  >
    <div
      className="sidebar__item-container selected"
      onClick={[Function]}
    >
      <div
        className="sidebar__item-icon"
      >
        <svg
          fill="white"
        >
          rooms.svg
        </svg>
      </div>
      <div>
        Rooms
      </div>
    </div>
    <div
      className="sidebar__item-container "
      onClick={[Function]}
    >
      <div
        className="sidebar__item-icon"
      >
        <svg
          fill="white"
        >
          monthly_schedule.svg
        </svg>
      </div>
      <div>
        Monthly Schedule
      </div>
    </div>
    <div
      className="sidebar__item-container "
      onClick={[Function]}
    >
      <div
        className="sidebar__item-icon"
      >
        <svg
          fill="white"
        >
          daily_schedule.svg
        </svg>
      </div>
      <div>
        Daily Schedule
      </div>
    </div>
  </div>
  <div
    className="app-layout__page-container"
  />
</div>
`;
